doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,453 @@ functionality over BLE and serial.
 Command-line Tool
 *****************
 
-MCUmgr provides a command-line tool, :file:`mcumgr`,for managing remote devices.
+MCUmgr provides a command-line tool, :file:`mcumgr`, for managing remote devices.
 The tool is written in the Go programming language.
 
-Installation information can be found in the `MCUmgr command-line tool`_
-section of the `MCUmgr documentation`_. A full rundown of the necessary commands
-to perform Device Firmware Upgrade is located in
-`MCUmgr command-line tool examples`_, and a practical step-by-step guide to DFU
-using MCUmgr can be found in :ref:`smp_svr_sample`.
+To install the tool::
+
+  go get github.com/apache/mynewt-mcumgr-cli/mcumgr
+
+Configuring the transport
+*************************
+
+There are two command-line options that are responsible for setting and configuring
+the transport layer to use when communicating with managed device:
+
+* ``--conntype`` is used to choose the transport used, and
+* ``--connstring`` is used to pass a comma separated list of options in the
+  ``key=value`` format, where each valid ``key`` depends on the particular
+  ``conntype``.
+
+Valid transports for ``--conntype`` are ``serial``, ``ble`` and ``udp``. Each
+transport expects a different set of key/value options:
+
+.. tabs::
+
+   .. group-tab:: serial
+
+      ``--connstring`` accepts the following ``key`` values:
+
+      * ``dev``: the device name for the OS ``mcumgr`` is running on (eg,
+        ``/dev/ttyUSB0``, ``/dev/tty.usbserial``, ``COM1``, etc).
+      * ``baud``: the communication speed; must match the baudrate of the server.
+      * ``mtu``: aka Maximum Transmission Unit, the maximum protocol packet size.
+
+   .. group-tab:: ble
+
+      ``--connstring`` accepts the following ``key`` values:
+
+      * ``ctlr_name``: An OS specific string for the controller name.
+      * ``own_addr_type``: Can be one of ``public``, ``random``, ``rpa_pub``,
+        ``rpa_rnd``, where ``random`` is the default.
+      * ``peer_name``: The name the peer BLE device advertises, this should match
+        the configuration specified with :option:`CONFIG_BT_DEVICE_NAME`.
+      * ``peer_id``: The peer BLE device address or UUID. Only required when
+        ``peer_name`` was not given. The format depends on the OS where ``mcumgr``
+        is run, it is a 6 bytes hexadecimal string separated by colons on Linux, or
+        a 128-bit UUID on macOS.
+      * ``conn_timeout``: A float number representing the connection timeout in
+        seconds.
```

Comment:
Style note: the letter casing after ':' is inconsistent.

Reply:
Fixed; I also organized the options a little better using tables.